src/main/java/uk/nhs/prm/repo/ehrtransferservice/repo_incoming/RepoIncomingService.java

```diff
@@ -15,15 +18,66 @@ public class RepoIncomingService {
 
     private static final String TRANSFER_TO_REPO_STARTED = "ACTION:TRANSFER_TO_REPO_STARTED";
     private static final String EHR_REQUEST_SENT = "ACTION:EHR_REQUEST_SENT";
-    private final SplunkAuditPublisher splunkAuditPublisher;
+
     private final TransferStore transferStore;
+    private final SplunkAuditPublisher splunkAuditPublisher;
     private final Gp2gpMessengerService gp2gpMessengerService;
 
+    @Value("${ehrResponsePollLimit}")
+    private int ehrResponsePollLimit;
+
+    @Value("${ehrResponsePollPeriodMilliseconds}")
+    private int ehrResponsePollPeriodMilliseconds;
+
     public void processIncomingEvent(RepoIncomingEvent repoIncomingEvent) throws Exception {
         boolean isActive = true;
         transferStore.createEhrTransfer(repoIncomingEvent, TRANSFER_TO_REPO_STARTED);
         splunkAuditPublisher.sendMessage(new SplunkAuditMessage(repoIncomingEvent.getConversationId(),repoIncomingEvent.getNemsMessageId(),TRANSFER_TO_REPO_STARTED));
         gp2gpMessengerService.sendEhrRequest(repoIncomingEvent);
         transferStore.handleEhrTransferStateUpdate(repoIncomingEvent.getConversationId(), repoIncomingEvent.getNemsMessageId(), EHR_REQUEST_SENT, isActive);
+        waitForTransferTrackerDbToUpdate(repoIncomingEvent.getConversationId());
+    }
+
+    private void waitForTransferTrackerDbToUpdate(String conversationId)
+            throws InterruptedException, EhrResponseFailedException, EhrResponseTimedOutException {
+        final float timeoutMinutes = ((float) ehrResponsePollLimit * ehrResponsePollPeriodMilliseconds) / (60 * 1000);
+        final float pollLimitSeconds = (float) ehrResponsePollPeriodMilliseconds / 1000;
+        int pollCount = 0;
+        String transferState = "";
+
+        log.info(String.format(
+                "Polling the TransferTrackerDB every %f seconds up to a maximum of %d times, this could take up to %f minutes",
+                pollLimitSeconds,
+                ehrResponsePollLimit,
+                timeoutMinutes));
+
+        while (pollCount < ehrResponsePollLimit) {
```

Comment:
Is this a candidate for a `do...while()` loop as we likely would like to check the status at least once?